rocketpy/mathutils/function.py

```diff
@@ -2612,6 +2622,77 @@ def compose(self, func, extrapolate=False):
                 extrapolation=self.__extrapolation__,
             )
 
+    def _check_user_input(
+        self,
+        source,
+        inputs,
+        outputs,
+        interpolation,
+        extrapolation,
+    ):
+        # check source for data type
+
+        # if list or ndarray, check for dimensions, interpolation and extrapolation
+        if isinstance(source, (list, np.ndarray)):
+            # this will also trigger an error if the source is not a list of numbers or if the array is not homogeneous
+            source = np.array(source, dtype=np.float64)
+
+            # check dimensions
+            source_dim = source.shape[1]
+
+            # check output type and dimensions
+            if isinstance(outputs, str):
+                outputs = [outputs]
+
+            elif len(outputs) > 1:
+                raise ValueError(
+                    f"Output must either be a string or have dimension 1, it currently has dimension ({len(outputs)})."
+                )
+
+            # check interpolation and extrapolation
+            if source_dim > 2:
+                # check for inputs and outputs
+                if inputs == ["Scalar"]:
+                    inputs = [f"Input {i+1}" for i in range(source_dim - 1)]
+                    warnings.warn(
+                        f"Inputs not set, defaulting to {inputs} for multidimensional functions.",
+                    )
+
+                if interpolation != None and interpolation != "shepard":
+                    interpolation = "shepard"
+                    warnings.warn(
+                        "Interpolation method for multidimensional functions is set to 'shepard', currently other methods are not supported.",
+                    )
+
+                if extrapolation == None:
+                    extrapolation = "natural"
+                    warnings.warn(
+                        "Extrapolation not set, defaulting to 'natural' for multidimensional functions.",
+                    )
+
+            # check input dimensions
+            in_out_dim = len(inputs) + len(outputs)
+            if source_dim != in_out_dim:
+                raise ValueError(
+                    f"Source dimension ({source_dim}) does not match input and output dimension ({in_out_dim})."
+                )
+
+        # if function, check for inputs and outputs
+        if isinstance(source, Function):
+            # check inputs
+            if inputs != None and inputs != source.get_inputs():
```

Comment:
Try to use "is not None" instead of "!= None", this improves readability in Python.

Reply:
Done